Formula/node.rb

```diff
@@ -107,12 +89,18 @@ def post_install
       # Dirs must exist first: https://github.com/Homebrew/legacy-homebrew/issues/35969
       mkdir_p HOMEBREW_PREFIX/"share/man/#{man}"
       rm_f Dir[HOMEBREW_PREFIX/"share/man/#{man}/{npm.,npm-,npmrc.,package.json.}*"]
-      ln_sf Dir[libexec/"npm/lib/node_modules/npm/man/#{man}/{npm,package.json}*"], HOMEBREW_PREFIX/"share/man/#{man}"
+      ln_sf Dir[libexec/"npm/man/#{man}/{npm,package.json}*"], HOMEBREW_PREFIX/"share/man/#{man}"
     end
 
     npm_root = node_modules/"npm"
     npmrc = npm_root/"npmrc"
     npmrc.atomic_write("prefix = #{HOMEBREW_PREFIX}\n")
+
+    # This is a minor hack that let's us keep a working npm
+    # in both HOMEBREW_PREFIX & also privately in libexec.
```

Comment:
You got there in roughly the same way I did:

```diff
diff --git a/Formula/node.rb b/Formula/node.rb
index 920e6233d2..220a35c4f4 100644
--- a/Formula/node.rb
+++ b/Formula/node.rb
@@ -40,7 +40,7 @@ class Node < Formula
   # We track major/minor from upstream Node releases.
   # We will accept *important* npm patch releases when necessary.
   resource "npm" do
-    url "https://registry.npmjs.org/npm/-/npm-5.0.1.tgz"
+    url "https://registry.npmjs.org/npm/-/npm-5.0.1.tgz", :using => :nounzip
     sha256 "cc31b845140791a85ab607636c6f0108346f8eb6cad7971919fd01ff4f4a1d7a"
   end
 
@@ -57,11 +57,22 @@ class Node < Formula
     system "make", "install"
 
     if build.with? "npm"
-      (libexec/"npm").install resource("npm")
+      # Allow npm to find Node before installation has completed.
+      ENV.prepend_path "PATH", bin
+
+      system "/usr/bin/tar", "-xvf", resource("npm").cached_download
+      system "#{buildpath}/package/bin/npm-cli.js", "install",
+             "--verbose", "--global", "--prefix=#{libexec}",
+             resource("npm").cached_download
+      # We never use these symlinks & they've caused issues in the past.
+      rm_rf libexec/"share"
+      # package.json has relative paths to the npm_install directory.
+      cp buildpath/"npm_install/package.json", libexec/"npm/lib/node_modules/npm"
+
 
       if build.with? "completion"
         bash_completion.install \
-          libexec/"npm/lib/utils/completion.sh" => "npm"
+          libexec/"lib/node_modules/npm/lib/utils/completion.sh" => "npm"
       end
     end
   end
@@ -75,7 +86,7 @@ class Node < Formula
     # Kill npm but preserve all other modules across node updates/upgrades.
     rm_rf node_modules/"npm"
 
-    cp_r libexec/"npm", node_modules
+    cp_r libexec/"lib/node_modules/npm", node_modules
     # This symlink doesn't hop into homebrew_prefix/bin automatically so
     # we make our own. This is a small consequence of our bottle npm &
     # retain a private copy in libexec setup. All other installs **do**
@@ -89,18 +100,12 @@ class Node < Formula
       # Dirs must exist first: https://github.com/Homebrew/legacy-homebrew/issues/35969
       mkdir_p HOMEBREW_PREFIX/"share/man/#{man}"
       rm_f Dir[HOMEBREW_PREFIX/"share/man/#{man}/{npm.,npm-,npmrc.,package.json.}*"]
-      ln_sf Dir[libexec/"npm/man/#{man}/{npm,package.json}*"], HOMEBREW_PREFIX/"share/man/#{man}"
+      ln_sf Dir[libexec/"npm/lib/node_modules/npm/man/#{man}/{npm,package.json}*"], HOMEBREW_PREFIX/"share/man/#{man}"
     end
 
     npm_root = node_modules/"npm"
     npmrc = npm_root/"npmrc"
     npmrc.atomic_write("prefix = #{HOMEBREW_PREFIX}\n")
-
-    # This is a minor hack that let's us keep a working npm
-    # in both HOMEBREW_PREFIX & also privately in libexec.
-    return if (libexec/"npm/bin/npm").read.include?(opt_libexec)
-    inreplace libexec/"npm/bin/npm", "$basedir/node_modules/npm/bin/npm-cli.js",
-                                     "#{opt_libexec}/npm/bin/npm-cli.js"
   end
 
   def caveats
```

I still think this is really pretty grim compared to the one-line `inreplace`.